lang/promise.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Allow all properties in type `T` to optionally have a {@link Promise} value.
+ *
+ * Use `await promised(obj) to translate a `Promised<T>` into a `T`.
+ */
+export type Promised<T extends Record<string, unknown>> = {
+  [P in keyof T]: T | Promise<T>;
+};
+
+/**
+ * Awaits the value for each property in an object and then returns a record with the values.
+ *
+ * @example
+ * You might resolve a record of file promises into a record of file contents.
+ * Before:
+ * ```
+ * {
+ *   'foo.txt': Promise<string>,
+ *   'foo.shasum': Promise<string>,
+ * }
+ * ```
+ * After:
+ * ```
+ * {
+ *   'foo.txt': 'a foo is a thing that does stuff',
+ *   'foo.shasum': 'c6feec83aa2ef60c68bb91d384c356dced2a3704e9769991be6b5baf4e382e91',
+ * }
+ * ```
+ *
+ * @example
+ * ```typescript
+ * function readFiles(...filePaths: string[]): Promise<Record<string, string>> {
+ *   const promisedFiles: Record<string, Promise<string>> = {};
+ *   filePaths.forEach(path => promisedFiles[path] = Deno.readTextFile(path));
+ *   return  (promisedFiles);
+ * }
+ * const fileContents = await readFiles('foo.txt', 'foo.shasum');
```

Comment:
Yeah, this was a mistake that I've since fixed. The return statement was meant to be `return promised(promisedFiles)`, but I accidentally left out the actual function call. 😅